app/services/subprocess_runner.py

```
@@ -31,22 +31,65 @@
 
 
 class ConversionCancelled(Exception):
     """Raised when a conversion is cancelled before completion."""
 
 
+# Hard bound on a filesystem probe taken on the spawn path, where there is no
+# stall loop yet to rescue a wait that never returns.
+_STAT_TIMEOUT = 10.0
+
+
+async def _bounded_probe(func, *args, **kwargs):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_bounded_probe`

(ANN202)